LayoutKit/Layouts/TextViewLayout.swift

```diff
@@ -0,0 +1,162 @@
+                  lineFragmentPadding: lineFragmentPadding,
+                  contentInset: contentInset,
+                  layoutMargins: layoutMargins,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    // MARK: - Layout protocol
+
+    open func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        let fittedSize = textSize(within: maxSize)
+        return LayoutMeasurement(layout: self, size: fittedSize.decreasedToSize(maxSize), maxSize: maxSize, sublayouts: [])
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    // MARK: - private helpers
+
+    private func textSize(within maxSize: CGSize) -> CGSize {
+        var size = TextCalculatorHelper.textSize(maxSize, text, font)
+
+        let heightInset = (textContainerInset.top + textContainerInset.bottom) + (layoutMargins.top + layoutMargins.bottom)
+        let widthInset = (textContainerInset.left + textContainerInset.right) + (layoutMargins.left + layoutMargins.right) + lineFragmentPadding * 2
+            + (contentInset.left + contentInset.right)
+        size = CGSize(width: size.width + widthInset, height: size.height + heightInset)
+
+        return size
+    }
+
+    // MARK: - overriden methods
+
+    open override func configure(view textView: TextView) {
+        config?(textView)
+        textView.textContainerInset = textContainerInset
+        textView.textContainer.lineFragmentPadding = lineFragmentPadding
+        textView.contentInset = contentInset
+        textView.layoutMargins = layoutMargins
+        textView.layoutManager.usesFontLeading = false
+        textView.font = font
+        switch text {
+        case .unattributed(let text):
+            textView.text = text
+        case .attributed(let attributedText):
+            textView.attributedText = attributedText
+        }
+    }
+
+    open override var needsView: Bool {
+        switch text {
+        case .attributed(let attributedText):
+            if attributedText.length == 0 {
+                return false
+            }
+        case .unattributed(let text):
+            if text.isEmpty {
+                return false
+            }
+        }
+
+        return true
+    }
+}
+
+// MARK: - Things that belong in TextViewLayout but aren't because TextViewLayout is generic.
+// "Static stored properties not yet supported in generic types"
+
+private let defaultFont: UIFont = {
+    let textView = UITextView()
+    // Font is only available when text has been set
+    textView.text = " "
+    return textView.font ?? UIFont.systemFont(ofSize: 17)
```

Comment:
Since we have to match the default font to `UITextView` and the system font for `UITextView` is dynamic selected based on iOS version.